crates/exec-wasmtime/src/loader/attested.rs

```diff
@@ -1,25 +1,116 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use super::{Acquired, Attested, Loader};
+use crate::Package;
 
-use std::{io::Read, os::unix::prelude::FromRawFd};
+use std::io::Read;
+use std::os::unix::prelude::FromRawFd;
 
-use anyhow::Result;
+use anyhow::{bail, ensure, Context, Result};
+use ureq::Response;
+
+/// Maximum size of WASM module in bytes
+const MAX_WASM_SIZE: u64 = 10_000_000;
+
+const DRAWBRIDGE_DIRECTORY_MEDIA_TYPE: &str = "application/vnd.drawbridge.directory.v1+json";
+const TOML_MEDIA_TYPE: &str = "application/toml";
+const WASM_MEDIA_TYPE: &str = "application/wasm";
+
+fn get(url: impl AsRef<str>) -> Result<Response> {
+    let url = url.as_ref();
+    ureq::get(url)
+        .call()
+        .with_context(|| format!("failed to GET `{}`", url))
+}
+
+fn response_into_wasm(res: Response) -> Result<Vec<u8>> {
+    // TODO: Initialize with capacity of Content-Length if set.
+    let mut wasm = Vec::new();
+    res.into_reader()
+        .take(MAX_WASM_SIZE)
+        .read_to_end(&mut wasm)
+        .context("failed to read WASM module contents")?;
+    // TODO: Verify Content-Digest
+    Ok(wasm)
+}
+
+fn get_drawbridge_directory(url: impl AsRef<str>) -> Result<(Vec<u8>, String)> {
+    let url = url.as_ref().trim_end_matches('/');
+    // TODO: refactor into `get_typed<WASM_MEDIA_TYPE>` once `&str` is supported as
+    // const generic argument
```

Comment:
What value would it bring?
It does not really help the particular issue, since enum variants cannot be used as generic arguments and we need strings anyway.
I briefly experimented with a trait containing an associated type, but that was too awkward, string is the simplest approach.